### PR TITLE
Removed support of v2.4.0 and added UT for v2.7.0

### PR DESCRIPTION
**Pull request by adarsh-dell**
# Description
Removed support of v2.4.0 and added UT for v2.7.0 and qualified v2.7.0 with cert-csi.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
|https://github.com/dell/csm/issues/743|

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Ran make unit-test.
image

- [x] Ran make driver-unit-test.
image

- [x] Installed csm-operator and driver v2.7.0 by using sample file provided for this version.
image

image

image

- [x] Ran cert-csi tool against v2.7.0 driver and all suites passed as per the expectation on time.
image

image

### CODE REVIEW

File: operatorconfig/driverconfig/powerstore/v2.7.0/controller.yaml [hunk 1]
```diff
@@ -131,7 +131,7 @@ spec:
             topologyKey: kubernetes.io/hostname 
       containers:
         - name: attacher
```
Comment (rensyct) [inline operatorconfig/driverconfig/powerstore/v2.7.0/controller.yaml:133]: I dont see support of 2.4 removed in this branch. Please remove that

File: tests/config/driverconfig/powerstore/v2.7.0/controller.yaml [hunk 2]
```diff
@@ -118,11 +127,11 @@ spec:
...
                   operator: In
                   values:
-                    - <DriverDefaultReleaseName>-controller
-            topologyKey: kubernetes.io/hostname
+                  - <DriverDefaultReleaseName>-controller
+            topologyKey: kubernetes.io/hostname 
       containers:
         - name: attacher
-          image: registry.k8s.io/sig-storage/csi-attacher:v3.5.0
+          image: k8s.gcr.io/sig-storage/csi-attacher:v4.2.0
```
Comment (rensyct) [inline tests/config/driverconfig/powerstore/v2.7.0/controller.yaml:134]: This should be registry.k8s.io. This was taken care for all links that were in repo earlier. Refer https://github.com/dell/csm-operator/pull/201

Please ensure to make this changes for all the files that are added new for PowerStore in this repo

File: tests/config/driverconfig/powerstore/v2.7.0/driver-config-params.yaml [hunk 3]
```diff
@@ -18,7 +18,7 @@ apiVersion: v1
 kind: ConfigMap
 metadata:
   name: <DriverDefaultReleaseName>-config-params
-  namespace: <DriverDefaultReleaseNamespace>
+  namespace: <DriverDefaultReleaseName>
```
Comment (rensyct) [inline tests/config/driverconfig/powerstore/v2.7.0/driver-config-params.yaml:21]: Should this not be DriverDefaultReleaseNamespace?

**Review by rensyct — CHANGES_REQUESTED** [review]

File: tests/config/driverconfig/powerstore/v2.7.0/driver-config-params.yaml (hunk 3 above)
Comment (adarsh-dell) [reply tests/config/driverconfig/powerstore/v2.7.0/driver-config-params.yaml]: Ran  _make driver-unit-test_
image

Ran _make unit_test_
image

File: tests/config/driverconfig/powerstore/v2.7.0/controller.yaml (hunk 2 above)
Comment (adarsh-dell) [reply tests/config/driverconfig/powerstore/v2.7.0/controller.yaml]: Pointed to use the suggested registry and installed the driver again.
image

Ran cert-csi again.
image

**Review by rajendraindukuri — APPROVED** [review]

**Pull request merged**